lnbits/decorators.py

```diff
@@ -49,185 +58,90 @@ def __init__(
                 name="X-API-KEY",
                 description="Wallet API Key - HEADER",
             )
-        self.wallet: Optional[Wallet] = None
         self.model: APIKey = openapi_model
 
-    async def __call__(self, request: Request):
+    async def __call__(self, request: Request) -> WalletTypeInfo:
         try:
+
             key_value = (
                 self._api_key
                 if self._api_key
                 else request.headers.get("X-API-KEY") or request.query_params["api-key"]
```

Comment:
A bit inconsistent:
 - if `request.query_params` is missing then an exception is raised and `HTTP 400 (BAD_REQUEST)` is returned
 - but if `request.query_params` is `None` or `""` then an `HTTP 401 (UNAUTHORIZED)` is returned

```suggestion
                else request.headers.get("X-API-KEY") or request.query_paramsget("api-key")
````

Reply:
as a consequence the catch block should catch a generic `Exception` and return an `Internal Server Error`